testsuite/shared/src/main/java/org/wildfly/test/jmx/ServiceActivatorDeploymentUtil.java

```diff
@@ -100,7 +111,15 @@ public static void createServiceActivatorListenerDeployment(File destination, St
                 getMBeanPermission(RunningStateJmx.class, targetName, "addNotificationListener"),
                 getMBeanPermission(RunningStateJmx.class, targetName, "removeNotificationListener"),
                 new PropertyPermission("user.dir", "read")
-        ), "permissions.xml");
+        ));
+        if(IS_IBM) {
+            //fix of WFCORE-3417
+            permissions.add(new FilePermission(destination.getAbsolutePath()
+                    .replace("archives", "domains/JmxControlledStateNotificationsTestCase/master")
```

Comment:
ServiceActivatorDeploymentUtil is an generic utility class, it should not mention something specific to JmxControlledStateNotificationsTestCase.
Instead, you could improve ServiceActivatorDeploymentUtil by adding a `createServiceActivatorListenerDeployment` helper method that would take an additional `Set<Permission>` so that tests that require additional permissions will be able to add them.

Reply:
It makes sense. I'll update PR to  reflect it. Thank you.